src/common/token.ts

```diff
@@ -167,19 +172,20 @@ export class ArweaveToken implements TokenTools<Transaction.default> {
   }
 
   public async submitTx({ tx }: { tx: Transaction.default }): Promise<void> {
-    const response = await this.arweave.transactions.post(tx).catch((err) => {
-      this.logger.error('Failed to post transaction...', { err });
-      return {
-        status: 500,
-        statusText:
-          err instanceof Error ? err.message : 'Failed to post Arweave Tx!',
-      };
-    });
-
-    if (response.status !== 200) {
+    try {
+      const response = await this.arweave.transactions.post(tx);
+
+      if (response.status !== 200) {
+        throw new Error(
+          'Failed to post transaction -- ' +
+            `Status ${response.status}, ${response.statusText}`,
+        );
+      }
```

Comment:
throw in a debug log if you find it helpful
```suggestion
      }
      this.logger.debug('Successfully posted fund transaction...', { tx });
```

Reply:
13237ad